### ISSUE TITLE
Creating postage stamps in parallel fails due to contract error

### ISSUE THREAD

**Issue by vojtechsimetka** [ethersphere/bee#1699]
#### Summary
When creating postage stamps in parallel, at most one such API call succeeds and other fails. This is due to the `approve` call for the BZZ token which is overwritten on each subsequent create postage stamp call. One potential issue that could arise is that due to the random order in which these parallel calls are processed, there may be some leftover approved BZZ balance if the batches are of different size. Not sure if this is exploitable.

#### Steps to reproduce
Run `curl -X POST -s localhost:1633/stamps/100000/20` at least twice in parallel.

#### Expected behavior
Ideally both of these calls should be successful (maybe on the API side added to some serial queue and processed one after the other). If that is not possible, then the API should reject and not try to process any new requests if there is already postage stamp being created. At some point adding some "create multiple different postage stamp batches" endpoint would be welcome.

#### Actual behavior

First terminal window
```
curl -X POST -s localhost:1633/stamps/100000/20
{"batchID":"f08e4f49b1700dc65691f044b590fd79f03735c0c319455dbaa9c1762ea29906"}
```
Second terminal window
```
curl -X POST -s localhost:1633/stamps/100000/20
{"message":"cannot create batch","code":500}
```
Ganache logs:
```
Block Number: 15176
Block Time: Wed May 12 2021 14:12:31 GMT+0000 (Coordinated Universal Time)
Transaction: 0xd5a24ff43ef6ac75b53c36f6ef837f0ec59935aa094020c5c66ea37fe781732c
Gas usage: 43598
Block Number: 15176
Block Time: Wed May 12 2021 14:12:31 GMT+0000 (Coordinated Universal Time)
Runtime Error: revert
Revert reason: ERC20: transfer amount exceeds allowance
```
Bee logs:
```
time="2021-05-12T14:15:10Z" level=info msg="api access" duration=16.9449622 ip=172.18.0.1 method=POST proto=HTTP/1.1 size=80 status=201 uri=/stamps/100/20 user-agent=curl/7.64.1
time="2021-05-12T14:15:10Z" level=debug msg="create batch: failed to create: VM Exception while processing transaction: revert ERC20: transfer amount exceeds allowance"
time="2021-05-12T14:15:10Z" level=error msg="create batch: failed to create"
time="2021-05-12T14:15:10Z" level=info msg="api access" duration=15.4130301 ip=172.18.0.1 method=POST proto=HTTP/1.1 size=46 status=500 uri=/stamps/100/20 user-agent=curl/7.64.1
```

**Comment by Eknir** [comment]
Thanks for reporting!

**Comment by acud** [comment]
this is now fixed with #2379
